ifExists function for MultiMap throws error

<!-- TEMPLATE -->

## Environment
OS: Linux Cinnamon
Node: v16.13.0
Google Chrome Version 105.0.5195.102 (Official Build) (64-bit)

<!-- Please specify how you run your code:
- OS: e.g. win/mac/linux
- runtime: e.g. Node 15.12.0 / Firefox 88.0.1
 -->

## Reproduction
I am following the example at [https://rimbu.org/api/rimbu/multimap/MultiMap/interface](https://rimbu.org/api/rimbu/multimap/MultiMap/interface) an dexperimenting.

I reproduce all examples up the modfyAt() function usage. The following example that uses the ifExists function does not compiles, with the error

```
const m = HashMultiMapHashValue.of([1, 'a'], [2, 'b'])
console.log(m.modifyAt(2, { ifExists: ['c'] }).toArray())
```

<!--
Consider creating a Codesandbox and supplying the link.
If you can't reproduce your bug, make sure to add details on how you've set up your project
-->

## Steps to reproduce

const m = HashMultiMapHashValue.of([1, 'a'], [2, 'b'])
console.log(m.modifyAt(2, { ifExists: ['c'] }).toArray())

## Expected Behavior

console output |  [[1, 'a'], [2, 'c']]

## Actual Behavior

Error: symptom-detail-history.component.ts:95:33 - error TS2322: Type 'string[]' is not assignable to type '(currentValues: NonEmpty<string>) => StreamSource<string>'.
  Type 'string[]' provides no match for the signature '(currentValues: NonEmpty<string>): StreamSource<string>'.

95     console.log(m.modifyAt(2, { ifExists: ['c'] }).toArray())
                                   ~~~~~~~~

  node_modules/.pnpm/@rimbu+multimap@0.9.19/node_modules/@rimbu/multimap/dist/types/custom/interface/base.d.ts:411:9
    411         ifExists?: (currentValues: WithKeyValue<Tp, K, V>['keyMapValuesNonEmpty']) => StreamSource<V>;
                ~~~~~~~~
    The expected type comes from property 'ifExists' which is declared here on type '{ ifNew?: OptLazy<StreamSource<string>>; ifExists?: (currentValues: NonEmpty<string>) => StreamSource<string>; }'




✖ Failed to compile.
-->

--- a/deno_dist/collection-types/map-custom/interface/index.ts
+++ b/deno_dist/collection-types/map-custom/interface/index.ts
@@ -387,11 +387,11 @@ export interface RMapBase<K, V, Tp extends RMapBase.Types = RMapBase.Types>
    * // => [[1, 'a'], [2, 'b'], [3, 'c']]
    * m.modifyAt(3, { ifNew: (none) => 1 < 2 ? none : 'c' }).toArray()
    * // => [[1, 'a'], [2, 'b']]
-   * m.modifyAt(2, { ifExists: 'c' }).toArray()
+   * m.modifyAt(2, { ifExists: () => 'c' }).toArray()
    * // => [[1, 'a'], [2, 'c']]
-   * m.modifyAt(2, { ifExists: v => v + 'z' }).toArray()
+   * m.modifyAt(2, { ifExists: (v) => v + 'z' }).toArray()
    * // => [[1, 'a'], [2, 'bz']]
-   * m.modifyAt(2, { ifExists: (v, remove) => v.length > 5 ? v : remove }).toArray()
+   * m.modifyAt(2, { ifExists: (v, remove) => v === 'a' ? v : remove }).toArray()
    * // => [[1, 'a']]
    * ```
    */
@@ -882,11 +882,11 @@ export namespace RMapBase {
      * // => true
      * m.modifyAt(3, { ifNew: (none) => 1 < 2 ? none : 'c' })
      * // => false
-     * m.modifyAt(2, { ifExists: 'c' })
+     * m.modifyAt(2, { ifExists: () => 'c' })
      * // => true
-     * m.modifyAt(1, { ifExists: v => v + 'z' })
+     * m.modifyAt(1, { ifExists: (v) => v + 'z' })
      * // => true
-     * m.modifyAt(2, { ifExists: (v, remove) => v.length > 5 ? v : remove })
+     * m.modifyAt(2, { ifExists: (v, remove) => v === 'a' ? v : remove })
      * // => true
      * ```
      */

--- a/deno_dist/graph/custom/valued/interface/generic/valued-graph-base.ts
+++ b/deno_dist/graph/custom/valued/interface/generic/valued-graph-base.ts
@@ -74,11 +74,11 @@ export interface ValuedGraphBase<
    * // => [[1, 2, 'a'], [2, 3, 'b'], [3, 4, 'c']]
    * g.modifyAt(3, 4, { ifNew: (none) => 1 < 2 ? none : 'c' }).toArray()
    * // => [[1, 2, 'a'], [2, 3, 'b']]
-   * g.modifyAt(1, 2, { ifExists: 'c' }).toArray()
+   * g.modifyAt(1, 2, { ifExists: () => 'c' }).toArray()
    * // => [[1, 2, 'c'], [2, 3, 'b']]
-   * g.modifyAt(1, 2, { ifExists: v => v + 'z' }).toArray()
+   * g.modifyAt(1, 2, { ifExists: (v) => v + 'z' }).toArray()
    * // => [[1, 2, 'az'], [2, 3, 'b']]
-   * g.modifyAt(2, 3, { ifExists: (v, remove) => v.length > 5 ? v : remove }).toArray()
+   * g.modifyAt(2, 3, { ifExists: (v, remove) => v === 'a' ? v : remove }).toArray()
    * // => [[1, 2, 'a']]
    * ```
    */
@@ -365,9 +365,9 @@ export namespace ValuedGraphBase {
      * b.modifyAt(3, 4, { ifNew: 'c' })                           // => true
      * g.modifyAt(4, 5, { ifNew: (none) => 1 < 2 ? none : 'c' })  // => false
      * g.modifyAt(1, 2, { ifNew: 'a' })                           // => false
-     * g.modifyAt(1, 2, { ifExists: 'c' })                        // => false
-     * g.modifyAt(1, 2, { ifExists: v => v + 'z' })               // => true
-     * g.modifyAt(2, 3, { ifExists: (v, remove) => v.length > 5 ? v : remove })
+     * g.modifyAt(1, 2, { ifExists: () => 'c' })                        // => false
+     * g.modifyAt(1, 2, { ifExists: (v) => v + 'z' })               // => true
+     * g.modifyAt(2, 3, { ifExists: (v, remove) => v === 'a' ? v : remove })
      * // => true
      * ```
      */

--- a/deno_dist/multimap/custom/interface/base.ts
+++ b/deno_dist/multimap/custom/interface/base.ts
@@ -463,11 +463,11 @@ export interface MultiMapBase<
    * // => [[1, 'a'], [2, 'b'], [3, 'c'], [3, 'd']]
    * m.modifyAt(3, { ifNew: () => 1 < 2 ? [] : ['c'] }).toArray()
    * // => [[1, 'a'], [2, 'b']]
-   * m.modifyAt(2, { ifExists: ['c'] }).toArray()
+   * m.modifyAt(2, { ifExists: () => ['c'] }).toArray()
    * // => [[1, 'a'], [2, 'c']]
-   * m.modifyAt(2, { ifExists: v => v.add('d') }).toArray()
+   * m.modifyAt(2, { ifExists: (v) => v.add('d') }).toArray()
    * // => [[1, 'a'], [2, 'c'], [2, 'd']]
-   * m.modifyAt(2, { ifExists: (v) => v.length > 3 ? v : [] }).toArray()
+   * m.modifyAt(2, { ifExists: (v) => v.has('a') ? v : [] }).toArray()
    * // => [[1, 'a']]
    * ```
    */

--- a/packages/collection-types/src/map-custom/interface/index.ts
+++ b/packages/collection-types/src/map-custom/interface/index.ts
@@ -387,11 +387,11 @@ export interface RMapBase<K, V, Tp extends RMapBase.Types = RMapBase.Types>
    * // => [[1, 'a'], [2, 'b'], [3, 'c']]
    * m.modifyAt(3, { ifNew: (none) => 1 < 2 ? none : 'c' }).toArray()
    * // => [[1, 'a'], [2, 'b']]
-   * m.modifyAt(2, { ifExists: 'c' }).toArray()
+   * m.modifyAt(2, { ifExists: () => 'c' }).toArray()
    * // => [[1, 'a'], [2, 'c']]
-   * m.modifyAt(2, { ifExists: v => v + 'z' }).toArray()
+   * m.modifyAt(2, { ifExists: (v) => v + 'z' }).toArray()
    * // => [[1, 'a'], [2, 'bz']]
-   * m.modifyAt(2, { ifExists: (v, remove) => v.length > 5 ? v : remove }).toArray()
+   * m.modifyAt(2, { ifExists: (v, remove) => v === 'a' ? v : remove }).toArray()
    * // => [[1, 'a']]
    * ```
    */
@@ -882,11 +882,11 @@ export namespace RMapBase {
      * // => true
      * m.modifyAt(3, { ifNew: (none) => 1 < 2 ? none : 'c' })
      * // => false
-     * m.modifyAt(2, { ifExists: 'c' })
+     * m.modifyAt(2, { ifExists: () => 'c' })
      * // => true
-     * m.modifyAt(1, { ifExists: v => v + 'z' })
+     * m.modifyAt(1, { ifExists: (v) => v + 'z' })
      * // => true
-     * m.modifyAt(2, { ifExists: (v, remove) => v.length > 5 ? v : remove })
+     * m.modifyAt(2, { ifExists: (v, remove) => v === 'a' ? v : remove })
      * // => true
      * ```
      */

--- a/packages/graph/src/custom/valued/interface/generic/valued-graph-base.ts
+++ b/packages/graph/src/custom/valued/interface/generic/valued-graph-base.ts
@@ -74,11 +74,11 @@ export interface ValuedGraphBase<
    * // => [[1, 2, 'a'], [2, 3, 'b'], [3, 4, 'c']]
    * g.modifyAt(3, 4, { ifNew: (none) => 1 < 2 ? none : 'c' }).toArray()
    * // => [[1, 2, 'a'], [2, 3, 'b']]
-   * g.modifyAt(1, 2, { ifExists: 'c' }).toArray()
+   * g.modifyAt(1, 2, { ifExists: () => 'c' }).toArray()
    * // => [[1, 2, 'c'], [2, 3, 'b']]
-   * g.modifyAt(1, 2, { ifExists: v => v + 'z' }).toArray()
+   * g.modifyAt(1, 2, { ifExists: (v) => v + 'z' }).toArray()
    * // => [[1, 2, 'az'], [2, 3, 'b']]
-   * g.modifyAt(2, 3, { ifExists: (v, remove) => v.length > 5 ? v : remove }).toArray()
+   * g.modifyAt(2, 3, { ifExists: (v, remove) => v === 'a' ? v : remove }).toArray()
    * // => [[1, 2, 'a']]
    * ```
    */
@@ -365,9 +365,9 @@ export namespace ValuedGraphBase {
      * b.modifyAt(3, 4, { ifNew: 'c' })                           // => true
      * g.modifyAt(4, 5, { ifNew: (none) => 1 < 2 ? none : 'c' })  // => false
      * g.modifyAt(1, 2, { ifNew: 'a' })                           // => false
-     * g.modifyAt(1, 2, { ifExists: 'c' })                        // => false
-     * g.modifyAt(1, 2, { ifExists: v => v + 'z' })               // => true
-     * g.modifyAt(2, 3, { ifExists: (v, remove) => v.length > 5 ? v : remove })
+     * g.modifyAt(1, 2, { ifExists: () => 'c' })                        // => false
+     * g.modifyAt(1, 2, { ifExists: (v) => v + 'z' })               // => true
+     * g.modifyAt(2, 3, { ifExists: (v, remove) => v === 'a' ? v : remove })
      * // => true
      * ```
      */

--- a/packages/multimap/src/custom/interface/base.ts
+++ b/packages/multimap/src/custom/interface/base.ts
@@ -463,11 +463,11 @@ export interface MultiMapBase<
    * // => [[1, 'a'], [2, 'b'], [3, 'c'], [3, 'd']]
    * m.modifyAt(3, { ifNew: () => 1 < 2 ? [] : ['c'] }).toArray()
    * // => [[1, 'a'], [2, 'b']]
-   * m.modifyAt(2, { ifExists: ['c'] }).toArray()
+   * m.modifyAt(2, { ifExists: () => ['c'] }).toArray()
    * // => [[1, 'a'], [2, 'c']]
-   * m.modifyAt(2, { ifExists: v => v.add('d') }).toArray()
+   * m.modifyAt(2, { ifExists: (v) => v.add('d') }).toArray()
    * // => [[1, 'a'], [2, 'c'], [2, 'd']]
-   * m.modifyAt(2, { ifExists: (v) => v.length > 3 ? v : [] }).toArray()
+   * m.modifyAt(2, { ifExists: (v) => v.has('a') ? v : [] }).toArray()
    * // => [[1, 'a']]
    * ```
    */

